R/derive_var_extreme_flag.R

```diff
@@ -196,53 +129,190 @@
 #'   "1023",    "SKIN DISORDERS",  "ERYTHEMA",   "MILD",       3,      1,
 #'   "1023",    "SKIN DISORDERS",  "ERYTHEMA", "SEVERE",       5,      2,
 #'   "1023",    "SKIN DISORDERS",  "ERYTHEMA",   "MILD",       8,      3
-#' )
+#' ) %>%
+#' mutate(STUDYID = "AB123")
+#'
+#' @caption Flagging the first/last observation within a by group
+#'
+#' @info A new variable is added for each subject to flag the last observation
+#'   within a by_group. Within each by group (specified by `by_vars`) the
+#'   `order = ADT` argument specifies we wish to sort the records by analysis
+#'   date and then select the last one (`mode = "last"`). The name of the new
+#'   variable is passed through the `new_var = LASTFL` call.
+#'
+#' @code
+#' advs %>%
+#'   derive_var_extreme_flag(
+#'     by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+#'     order = exprs(ADT),
+#'     new_var = LASTFL,
+#'     mode = "last",
+#'   ) %>%
+#'   arrange(STUDYID, USUBJID, PARAMCD, ADT) %>%
+#'   select(STUDYID, everything())
+#'
+#' @info Note here that a similar `FIRSTFL` variable could instead be derived
+#'    simply by switching to `mode = "first"`. Alternatively, we could make use
+#'    of `desc()` within the sorting specified by `order`:
 #'
-#' # Most severe AE first occurrence per patient
-#' example_ae %>%
+#' @code
+#' advs %>%
+#'   derive_var_extreme_flag(
+#'     by_vars = exprs(STUDYID, USUBJID, PARAMCD),
+#'     order = exprs(desc(ADT)),
+#'     new_var = FIRSTFL,
+#'     mode = "last",
+#'   ) %>%
+#'   arrange(STUDYID, USUBJID, PARAMCD, ADT) %>%
+#'   select(STUDYID, everything())
+#'
+#' @caption Modifying the flag values (`true_value`, `false_value`)
+#'
+#' @info The previous example is now enhanced with custom values for the flag
+#'   entries. Records which are flagged are filled with the contents of
+#'   `true_value` and those which are not are filled with the contents of
```

Comment:
```suggestion
#'   `true_value` and the other records are filled with the contents of
```
i was just trying to make this read a little clearer, but feel free to ignore